HW10/classes.js

```diff
@@ -0,0 +1,171 @@
+/** Завдання створити класи:
+ *    фігура:
+ *      координати:
+ *      х, у
+ *
+ *    багатокутник -> фігура:
+ *      вершини[координати],
+ *      периметр()
+ *
+ *    прямокутник -> багатокутник:
+ *      довжина,
+ *      висота,
+ *      площа(),
+ *      периметр()
+ *
+ *    квадрат -> прямокутник:
+ *      довжинаСторони,
+ *      площа(),
+ *      периметр()
+ *
+ *    круг -> фігура:
+ *      радіус,
+ *      площа(),
+ *      периметр()
+ */
+
+class Shape {
+  /**
+   * @param {array} center
+   */
+  constructor(center) {
+    this.center = center;
+  }
+}
+
+/* багатокутник -> фігура:
+ *      вершини[координати],
+ *      периметр() */
+class Polygon extends Shape {
+  /**
+  * @param {array} center
+  * @param {object} points
+  * */
+  constructor(center, points) {
+    super(center);
+    this.points = points;
+  }
+}
+
+/* прямокутник -> багатокутник:
+ *      довжина,
+ *      висота,
+ *      площа(),
+ *      периметр()
+ */
+class Rectangle extends Polygon {
+  /**
+   * @param {array} center
+   * @param {number} width
+   * @param {number} height
+   * */
+  constructor(center, width, height) {
+    super(center);
+    this.width = width;
+    this.height = height;
+  }
+
+  square() {
+    /**
+     * return number
+     */
+    return this.width * this.height;
+  }
+
+  perimeter() {
+    /**
+     * return number
```

Comment:
```suggestion
     * @returns {number}
```
http://usejsdoc.org/tags-returns.html